src/components/element-editor/element-editor.tsx

```diff
@@ -0,0 +1,124 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Button, Form } from "@lifesg/react-design-system";
+import { useMemo } from "react";
+import { FormProvider, useForm, useFormContext } from "react-hook-form";
+import { useBuilder } from "src/context-providers";
+import * as yup from "yup";
+import { IFieldValues } from "./types";
+
+export const ElementEditor = () => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const { elements, focusedElement } = useBuilder();
+
+    const camelCaseRegex = /^[a-z]+((\d)|([A-Z0-9][a-z0-9]+))*([A-Z])?$/gm;
+    const schema = useMemo(() => {
+        return yup.object<IFieldValues>().shape({
+            id: yup
+                .string()
+                .required("ID is required")
+                .matches(camelCaseRegex, { message: "ID must be camelcase" })
+                .notOneOf(
+                    Array.from(elements.values())
+                        .map((e) => e.id)
+                        .filter((id) => id !== focusedElement.element.id),
+                    "ID must not be duplicated"
+                ),
+            label: yup.string().required(),
+            type: yup.string().required(),
+            placeholder: yup.string().optional(),
+        });
+    }, []);
+
+    const methods = useForm({
+        mode: "onTouched",
+        resolver: yupResolver(schema),
+    });
+
+    // =========================================================================
```

Comment:
Remove unnecessary section headers. There are no functions in them. Add when necessary.